RealestateApi/Service/AgentService.cs

```diff
@@ -36,5 +36,23 @@ public async Task<ServiceResult<AgentModel>> GetAgentByIdAsync(int agentId)
         {
             return await _agentRepository.GetAgentByIdAsync(agentId);
         }
+
+        public async Task<ServiceResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
+        {
+            ServiceResult<AgentModel> response = new();
+
+            var existAgent = await GetAgentByIdAsync(agentId);
+
+            if (!existAgent.IsSuccess)
+            {
+                response.AdditionalInformation.Add($"Agent with ID {agentId} doesn't exist");
+                return response;
+            }
+
+            response = await _agentRepository.DeleteAgentByIdAsync(agentId);
```

Comment:
Response is not being validated.

IsSuccess should be done against response and not against existAgent.

Reply:
Done